### PR TITLE
Fixed #31523 -- Removed jQuery dependency from actions.js.

### PR DESCRIPTION
**Pull request by jdufresne**
https://code.djangoproject.com/ticket/31523

### CODE REVIEW

**Comment by jdufresne** [comment]
This more or less amounts to a rewrite of `actions.js`. If it would assist with review, I can try to break down the change into incremental compatible steps. If that is preferred, just let me know.

File: django/contrib/admin/static/admin/js/actions.js [hunk 1]
```diff
@@ -1,153 +1,171 @@
...
-    };
-    $(document).ready(function() {
-        var $actionsEls = $('tr input.action-select');
-        if ($actionsEls.length > 0) {
-            $actionsEls.actions();
+
+    // Call function fn when the DOM is loaded and ready. If it is already
+    // loaded, call the function now.
+    // http://youmightnotneedjquery.com/#ready
+    function ready(fn) {
```
Comment (jdufresne) [inline django/contrib/admin/static/admin/js/actions.js:157]: This same function also exists in `django/contrib/admin/static/admin/js/cancel.js`. If you think this duplication should be removed by adding a new common function, please let me know.

**Comment by jdufresne** [comment]
buildbot, test on selenium.

**Comment by carltongibson** [comment]
I'd like to take #12819 before this.

**Comment by carltongibson** [comment]
Hey @jdufresne. If you could rebase, I shall take it on after the weekend. Thanks!

**Comment by jdufresne** [comment]
buildbot, test on selenium.

**Comment by jdufresne** [comment]
Rebased :slightly_smiling_face:

**Comment by carltongibson** [comment]
buildbot, test on selenium.

**Comment by carltongibson** [comment]
CI failure is unrelated.

**Pull request merged**